astropy/coordinates/erfa_astrom.py

```diff
@@ -0,0 +1,269 @@
+
+    @staticmethod
+    def apio13(frame_or_coord):
+        '''
+        Wrapper for ``erfa.apio13``, used in conversions AltAz <-> CIRS
+
+        Arguments
+        ---------
+        frame_or_coord: ``astropy.coordinates.BaseCoordinateFrame`` or ``astropy.coordinates.SkyCoord``
+            Frame or coordinate instance in the corresponding frame
+            for which to calculate the calculate the astrom values.
+            For this function, an AltAz frame is expected.
+        '''
+        lon, lat, height = frame_or_coord.location.to_geodetic('WGS84')
+
+        jd1_utc, jd2_utc = get_jd12(frame_or_coord.obstime, 'utc')
+        dut1utc = get_dut1utc(frame_or_coord.obstime)
+
+        return erfa.apio13(
+            jd1_utc, jd2_utc, dut1utc,
+            lon.to_value(u.radian),
+            lat.to_value(u.radian),
+            height.to_value(u.m),
+            *get_polar_motion(frame_or_coord.obstime),
+            # all below are already in correct units because they are QuantityFrameAttribues
+            frame_or_coord.pressure.value,
+            frame_or_coord.temperature.value,
+            frame_or_coord.relative_humidity.value,
+            frame_or_coord.obswl.value,
+        )
+
+
+class ErfaAstromInterpolator(ErfaAstrom):
+    '''
+    A provider for astrometry values that does not call erfa
+    for each individual timestamp but interpolates linearly
+    between support points.
+
+    For the interpolation, float64 MJD values are used, so time precision
+    for the interpolation will be around a microsecond.
+
+    This can dramatically speed up coordinate transformations,
+    e.g. between CIRS and ICRS,
+    when obstime is an array of many values (factors of 10 to > 100 depending
+    on the selected resolution, number of points and the time range of the values).
+
+    The precision of the transformation will still be in the order of microseconds
+    for reasonable values of time_resolution, e.g. ``300 * u.s``.
+
+    Users should benchmark performance and accuracy with the default transformation
+    for their specific use case and then choose a suitable ``time_resolution``
+    from there.
+
+    This class is intended be used together with the ``erfa_astrom`` science state,
+    e.g. in a context manager like this
+
+    Example
+    -------
+    >>> from astropy.coordinates import SkyCoord, CIRS
+    >>> from astropy.coordinates.erfa_astrom import erfa_astrom, ErfaAstromInterpolator
+    >>> import astropy.units as u
+    >>> from astropy.time import Time
+    >>> import numpy as np
+
+    >>> obstime = Time.now() + np.linspace(0, 4, 1000) * u.hour
+    >>> with erfa_astrom.set(ErfaAstromInterpolator(300 * u.s)):
+    ...    cirs = SkyCoord.from_name('Crab').transform_to(CIRS(obstime=obstime))  # doctest: +REMOTE_DATA
+    '''
+
+    @u.quantity_input(time_resolution=u.day)
+    def __init__(self, time_resolution):
+        if time_resolution.to_value(u.us) < 10:
```

Comment:
It's a factor of > 4 slower. I see that is not a performance bottleneck here, but we made the experience that one should use the best practices as people reading the code adapt usage they see. 

This is also the reason for using `u.Quantity(value, unit)` instead of `value * unit` below.

Again, I see this is not performance critical here but I don't see a real legibility improvement through the other, worse performing notation.

Reply:
What about `value << unit` then?

Reply:
I don't see how the `<<` operator would help in this comparison.

Reply:
Also, `<<` is probably the worst option when it comes to readability.

Reply:
I suggested it hoping that it would be slightly faster than `value * unit`, but the performance gain [of this trick]() is only visible when working with large arrays.

I can confirm that your version is faster:

```python
In [11]: %timeit time_resolution < 10 * u.us                                                                                                                                                                                                  
22 µs ± 357 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [12]: %timeit time_resolution < (10 << u.us)                                                                                                                                                                                               
22.4 µs ± 214 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [13]: %timeit time_resolution.to_value(u.us) < 10                                                                                                                                                                                          
1.11 µs ± 10.3 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```